### ISSUE TITLE
Spin App won't connect

### ISSUE THREAD

**Issue by semiconducted** [ttseng/Spin#2]
I am building this using a DIY Easy Driver shield and have it breadboarded at the moment. Everything seems to be wired correctly, using the 'patched' Softmodem library, the multimeter reads the pins from the TRRS iphone cable correctly, that circuit seems to be correct as well.

When I click begin setup > begin > next > 
It says "connecting to turntable...", spins the stepper motor around a few times, but then the iOS app says "unable to connect turntable!".

I've gone over everything for days and can't figure out why this won't work. Any ideas?

Thanks!

- Damon, aka Semiconducted

**Comment by ttseng** [comment]
Hi, there!  I'll do my best to help debug.  Let's figure out if it's a hardware issue with the turntable itself first.  Are you able to successfully run the stepper motor via USB from the Arduino with the Processing sketch here? http://spin.media.mit.edu/build#program_arduino

**Comment by semiconducted** [comment]
Thanks for your response!

Yes, the processing sketch works. I had to crank up the motor_speed to 1 and multiply rotAmt by 10 to see any effect, but yes. 
The iOS app throws the error, right after the stepper does its little dance. So it seems the SoftModem is issuing commands, but the board is not sending or the app is not recognizing the feedback? I see numbers popping in the Serial Monitor, but not sure what to look for or how to troubleshoot.

Also, is it normal for the stepper motor to be loud and vibratey? Reducing vibration and noise would be a plus, as my end goal is to customize Spin for 3D scanning. If you know anyone who’s taken that direction, let me know! 
I started investigating replacing the motor control methods with the AccelStepper library to hopefully make it smoother and more extensible, but software might not actually be what’s cause that other issue.

Thanks!

- Damon
____
Damon Hamm
www.semiconducted.com
d@damonhamm.com
www.linkedin.com/in/damonhamm

**Comment by ttseng** [comment]
After the app sends a signal to the turntable, it expects to receive a signal back on the mic line of the audio cable.  I'm using the ring closest to the base of the audio cable for the mic line, so you might try ensuring that both ends of the 4-pin audio jack are connected at that ring (I have experienced purchasing faulty 4 pin audio cables in the past).  It's also possible that the Arduino is sending a signal via the mic line back to the phone but it's not being amplified enough for the phone to detect.  I don't know if having a different motor driver might affect the signal, but you could try playing around with R2 and R3 to ensure the signal is strong enough.

If you have a custom version of the Easy Driver, it's important to ensure that it's capable of microstepping if you care about volume  I've tried using other motor drivers (TB6612FNG in particular) and it was extremely loud since it couldn't microstep.

Let me know if that helps!

**Comment by semiconducted** [comment]
Still no luck changing the resistors to the original SoftModem circuit below and also changed R2 and R3 both to 4.7k, to no avail.
I’m curious, why is your schematic for this circuit different than the creators'? What are the advantages?

I’m not sure how exactly to test or if I should change the resistors up or down. My multimeter shows a signal coming through all the way to the ring (it is a little exposed), though the app does not respond. In the meantime I ordered the exact cable you specified from Amazon.com in the hopes it is my cable.

Thanks!

- Damon

 https://raw.githubusercontent.com/arms22/SoftModem/master/extras/schematic.png <https://raw.githubusercontent.com/arms22/SoftModem/master/extras/schematic.png>

**Comment by semiconducted** [comment]
Hey Tiffany, Damon Hamm here just following up with an update.I was able to track down what was causing the problem and of course it was the last place I looked - a bad motor power supply! It was registering correct voltage and rated for 3A, but when running the motor the motor was insanely choppy and loud. I of course thought it was the motor driver chip, something else in the circuit, yada yada. It was only when I used an oscilloscope to analyze the audio signals that I discovered the noise that obliterated everything else. I swapped out the motor power supply and suddenly everything worked perfectly.Thanks again for sharing this project! I plan to post my DIY shield setup and arduino code mods when finished. 
- Damon HammP.S. I also noticed you also are a UXer! Since the MIT media lab wouldn’t have me ☹️, I did my masters at Carnegie Mellon 😀.
www.damonhamm.com
www.semiconducted.com
